lib/awful/hotkeys_popup/widget.lua

```diff
@@ -141,53 +142,79 @@ function widget.new(args)
         ) and widget.merge_duplicates or args.merge_duplicates,
         group_rules = args.group_rules or gtable.clone(widget.group_rules),
         labels = args.labels or {
-            Mod4="Super",
-            Mod1="Alt",
-            Escape="Esc",
-            Insert="Ins",
-            Delete="Del",
-            Backspace="BackSpc",
-            Return="Enter",
-            Next="PgDn",
-            Prior="PgUp",
-            ['#108']="Alt Gr",
-            Left='←',
-            Up='↑',
-            Right='→',
-            Down='↓',
-            ['#67']="F1",
-            ['#68']="F2",
-            ['#69']="F3",
-            ['#70']="F4",
-            ['#71']="F5",
-            ['#72']="F6",
-            ['#73']="F7",
-            ['#74']="F8",
-            ['#75']="F9",
-            ['#76']="F10",
-            ['#95']="F11",
-            ['#96']="F12",
-            ['#10']="1",
-            ['#11']="2",
-            ['#12']="3",
-            ['#13']="4",
-            ['#14']="5",
-            ['#15']="6",
-            ['#16']="7",
-            ['#17']="8",
-            ['#18']="9",
-            ['#19']="0",
-            ['#20']="-",
-            ['#21']="=",
-            Control="Ctrl"
+            Mod4       ="Super",
```

Comment:
This should be ported to C, the map is incomplete and doesn't support everything. We already have such mechanism for the modkeys, see awesome._modifiers (`luaa.c` `static int luaA_get_modifiers(lua_State *L)`).